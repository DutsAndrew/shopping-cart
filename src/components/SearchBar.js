import React from "react";
import '../style/SearchBar.css';

const SearchBar = props => {

  const { submit } = props;

  return (
    <form id="search-form">
      <input id="search-input" type="text" placeholder="Elden Ring" ></input>
      <button id="search-button" type="submit" onClick={submit} >Find Game</button>
    </form>
  );
};

export default SearchBar;