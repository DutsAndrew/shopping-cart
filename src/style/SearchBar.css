#search-form {
  display: flex;
  justify-content: center;
  align-items: center;
  gap: 1%;
  width: 100vw;
}

#search-input {
  font-size: 1.5em;
}

#search-input {
  border-radius: 5%;
}

#search-button {
  height: 3vh;
  width: 10vw;
  font-size: 1em;
  background-color: #66c0f4;
  color: #1b2838;
}