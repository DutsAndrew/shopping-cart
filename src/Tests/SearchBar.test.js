import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import SearchBar from '../components/SearchBar';

describe('SearchBar component', () => {
  it('renders search form', () => {
    const { container } = render(<SearchBar />);
    expect(container).toMatchSnapshot();
  });

  it('submits form', () => {
    const onSubmitMock = jest.fn();
    render(<SearchBar submit={onSubmitMock} />);
    const button = screen.getByRole('button');
    userEvent.click(button);
    expect(onSubmitMock).toBeCalledTimes(1);
  });
});