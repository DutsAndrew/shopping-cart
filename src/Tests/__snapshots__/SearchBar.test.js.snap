// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar component renders search form 1`] = `
<div>
  <form
    id="search-form"
  >
    <input
      id="search-input"
      placeholder="Elden Ring"
      type="text"
    />
    <button
      id="search-button"
      type="submit"
    >
      Find Game
    </button>
  </form>
</div>
`;
